### PR TITLE
Fix bug to autofocus okay button on overvote modal

### PR DESCRIPTION
**Pull request by carolinemodic**
## Overview
https://github.com/votingworks/vxsuite/issues/4927 
Because of the way React automatically handles focusing of Modals using the autofocus html5 attribute on a button inside a Modal does not work and the "Okay" to dismiss the overvote warning modal does not get properly autofocused. Instead we have to manually focus the button ref in order for it to autofocus as implemented here. Note the onAfterOpen was not triggering at the right time in the Modal component unless I passed through the isOpen parameter I know we don't otherwise use. 

## Testing Plan
Manually triggered overvotes saw "okay" focus 

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Review by kofi-q — COMMENTED** [review]
This approach might cancel out the screen reader audio that's supposed to play when the modal opens (so, instead of reading out the overvote message, it just reads the button audio instead).

I think the more complete/pattern-consistent fix would be to update the audio message to say `"Press the right button to continue"` instead of `"Press the select button to continue"` and then assign the ["next" navigation id](https://github.com/votingworks/vxsuite/blob/273a2d9/libs/ui/src/language_settings/language_settings_screen.tsx#L118) to the Okay button. The issue there is that the old navigation logic assumes theres only one "next" id button on the page at a time, but the contest page underneath the modal also has a "next" button. We'd need to update the [navigation logic](https://github.com/votingworks/vxsuite/blob/8565b9a/libs/ui/src/accessible_controllers/navigation.ts#L60) to use `querySelectorAll` instead of `getElementById` and filter out the hidden one with the [`getTabEnabledElementsInHiddenBlocks`](https://github.com/votingworks/vxsuite/blob/8565b9a/libs/ui/src/accessible_controllers/navigation.ts#L20) utility.

**Comment by carolinemodic** [comment]
Ah you're right good catch

File: libs/mark-flow-ui/src/components/candidate_contest.tsx [hunk 1]
```diff
@@ -392,6 +393,7 @@ export function CandidateContest({
               variant="primary"
               autoFocus
               onPress={closeAttemptedVoteAlert}
+              id={PageNavigationButtonId.NEXT}
             >
               {appStrings.buttonOkay()}
               <AudioOnly>
```
Comment (carolinemodic) [inline libs/mark-flow-ui/src/components/candidate_contest.tsx:399]: Kept these next audio descriptions unchanged as at that point you are focusing on the Okay button

**Comment by carolinemodic** [comment]
back to you @kofi-q

**Review by kofi-q — APPROVED** [review]
LGTM!

**Pull request merged**